ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -1,28 +1,49 @@
-{-# LANGUAGE FlexibleContexts    #-}
-{-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE DeriveGeneric        #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE NamedFieldPuns       #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE StandaloneDeriving   #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE UndecidableInstances #-}
 
--- | Monadic side of the Mempool implementation.
---
--- Using the functions defined in Ouroboros.Consensus.Mempool.Impl.Pure,
--- a dedicated constructor 'openMempool' is provided to encapsulate the mempool
--- functionality.
---
--- The implementation is based on a MempoolEnv that captures the relevant
--- variables to manage the mempool and is then used to craft functions that
--- conform to the Mempool datatype API.
---
--- The operations performed on the Mempool are written in a pure fashion in
--- Ouroboros.Consensus.Mempool.Impl.Pure.
+-- | Definition of common types used in "Ouroboros.Consensus.Mempool.Init",
```

Comment:
It's a bit strange that we do not mention anymore the fact that this module implements the mempool. If it doesn't anymore maybe this module should be renamed to something like `*.Common`.

Reply:
See https://github.com/input-output-hk/ouroboros-network/pull/4332#discussion_r1098682795